client/components/Editor/plugins/discussions/util.ts

```diff
@@ -32,15 +32,23 @@ export const mapDiscussionSelectionThroughSteps = (
 	if (isEmptySelection(selection)) {
 		return null;
 	}
-	const mapping = new Mapping(steps.map((step) => step.getMap()));
 	const { from, to } = getRangeFromSelection(selection!);
 	// This matches the default mapping of inline Decorations, which we wish to emulate here.
 	// The default Selection mapping behavior is too greedy; for instance, if a selection is made
 	// at the beginning of the document, any text inserted before that selection is also mapped
 	// into that selection, which is not what we want for anchored Discussions.
 	// See: https://github.com/ProseMirror/prosemirror-view/blob/49889c576bc450a3008117249f9ef9beb6bf2969/src/decoration.js#L36
-	const nextFrom = mapping.map(from, 1);
-	const nextTo = mapping.map(to, -1);
+	const { from: nextFrom, to: nextTo } = steps.reduce(
+		(state, step) => {
+			const { from: currentFrom, to: currentTo } = state;
+			const stepMap = step.getMap();
+			return {
+				from: stepMap.map(currentFrom, -1),
```

Comment:
Is this replicating the functionality described in the comment block above? And if so, is the -1 intentional (The removed line had a positive value for from: `const nextFrom = mapping.map(from, 1);`

Reply:
Good eye, that would have been a nightmare to track down

Reply:
👍 Testing it out, I can see that it produced the wrong behavior too - so we have a nice condition to validate against once this fix lands on qubqub. This comment was made when "Hello" was the first line in the doc, and adding content before it shows the mapping being off.
<img width="908" alt="Screen Shot 2021-02-01 at 3 53 05 PM" src="https://user-images.githubusercontent.com/1000455/106517135-fc5ea100-64cf-11eb-82da-e96b57bbcc38.png">

Reply:
++ I added a unit test case for this as well